integration-tests/cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/configuration/ReplaceKeys.feature

```diff
@@ -17,14 +16,36 @@ Feature: SmartMetering Configuration - Replace Keys
     Then the replace keys response should be returned
       | DeviceIdentification | TEST1024000000001 |
       | Result               | OK                |
-    And the new keys are stored in the osgp_adapter_protocol_dlms database security_key table
+    And the new keys are stored in the secret management database encrypted_secret table
     And the stored keys are not equal to the received keys
 
   @ResetKeysOnDevice
   Scenario: Replace keys with generated ones on a device
+    Given a dlms device
+      | DeviceIdentification | TEST1024000000001 |
+      | DeviceType           | SMART_METER_E     |
     When the generate and replace keys request is received
       | DeviceIdentification | TEST1024000000001 |
     Then the generate and replace keys response should be returned
       | DeviceIdentification | TEST1024000000001 |
       | Result               | OK                |
-    And the new keys are stored in the osgp_adapter_protocol_dlms database security_key table
+    And the new keys are stored in the secret management database encrypted_secret table
+
+  @RecoverKeys
```

Comment:
Are these Key specific annotations intentionally?

Reply:
Yes, to be able to test key recovery specifically